File: paddler_types/src/conversation_message.rs
use serde::Deserialize;
use serde::Serialize;

use crate::conversation_message_content::ConversationMessageContent;

#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
#[serde(deny_unknown_fields)]
pub struct ConversationMessage {
    pub content: ConversationMessageContent,
    pub role: String,
}

File: paddler_client_python/paddler_client/types/conversation_message.py
from pydantic import BaseModel

from paddler_client.types.conversation_message_content_part import (
    ConversationMessageContentPart,
)

ConversationMessageContent = str | list[ConversationMessageContentPart]


class ConversationMessage(BaseModel):
    content: ConversationMessageContent
    role: str
